### DISCUSSION TITLE
Externalise native plugin description

### DISCUSSION THREAD

**Discussion by Manish-22** [microsoft/semantic-kernel#7058] [Q&A] [answered]
For a native function such as this 

```
[KernelFunction("get_lights")]
[Description("Gets a list of lights and their current state")]
[return: Description("An array of lights")]
public async Task<List<LightModel>> GetLightsAsync()
{
   return lights
}
```

How can I set the description from a text file or a database?
Would be helpful as it would be easy to change description without redeployment of code

**Comment by dmytrostruk** [comment, marked as answer]
@Manish-22 You can use imperative approach how to define and import native functions in Kernel. Based on your example, it will look like this:
```csharp
kernel.ImportPluginFromFunctions("MyPlugin",
[
    kernel.CreateFunctionFromMethod(
        () => lights,
        functionName: "get_lights",
        description: "Gets a list of lights and their current state",
        returnParameter: new() { Description = "An array of lights" }),
]);
```

For description, you can call another service which will pull your description from text file, configuration or a database. Let us know if that covers your scenario. Thanks!

**Reply by Manish-22** [reply]
Thanks @dmytrostruk
